storage/database/db_manager_test.go

```diff
@@ -0,0 +1,314 @@
+
+		dbm.WriteHeadFastBlockHash(hash2)
+		assert.Equal(t, hash2, dbm.ReadHeadFastBlockHash())
+	}
+}
+
+// TestDBManager_FastTrieProgress tests read and write operations of fast trie progress.
+func TestDBManager_FastTrieProgress(t *testing.T) {
+	for _, dbm := range dbManagers {
+		assert.Equal(t, uint64(0), dbm.ReadFastTrieProgress())
+
+		dbm.WriteFastTrieProgress(num1)
+		assert.Equal(t, num1, dbm.ReadFastTrieProgress())
+
+		dbm.WriteFastTrieProgress(num2)
+		assert.Equal(t, num2, dbm.ReadFastTrieProgress())
+	}
+}
+
+func TestDBManager_Header(t *testing.T) {
+	// TODO-Klaytn-Database Implement this!
+}
+
+func TestDBManager_Body(t *testing.T) {
+	// TODO-Klaytn-Database Implement this!
+}
+
+func TestDBManager_Td(t *testing.T) {
+	// TODO-Klaytn-Database Implement this!
+}
+
+func TestDBManager_Receipts(t *testing.T) {
+	// TODO-Klaytn-Database Implement this!
+}
+
+func TestDBManager_Block(t *testing.T) {
+	// TODO-Klaytn-Database Implement this!
+}
+
+func TestDBManager_IstanbulSnapshot(t *testing.T) {
+	// TODO-Klaytn-Database Implement this!
+}
+
+// TestDBManager_DatabaseVersion tests read/write operations of database version.
+func TestDBManager_DatabaseVersion(t *testing.T) {
+	// TODO-Klaytn-Database DatabaseVersion should be handled carefully.
+	//for i, dbm := range dbManagers {
+	//	c := dbConfigs[i]
+	//
+	//	assert.Equal(t, uint64(0), dbm.ReadDatabaseVersion())
+	//
+	//	dbm.WriteDatabaseVersion(uint64(1))
+	//	assert.Equal(t, uint64(1), dbm.ReadDatabaseVersion())
+	//
+	//	dbm.WriteDatabaseVersion(uint64(2))
+	//	assert.Equal(t, uint64(2), dbm.ReadDatabaseVersion())
+	//}
+}
+
+// TestDBManager_ChainConfig tests read/write operations of chain configuration.
+func TestDBManager_ChainConfig(t *testing.T) {
+	for _, dbm := range dbManagers {
+		assert.Nil(t, nil, dbm.ReadChainConfig(hash1))
+
+		cc1 := &params.ChainConfig{UnitPrice: 12345}
+		cc2 := &params.ChainConfig{UnitPrice: 54321}
+
+		dbm.WriteChainConfig(hash1, cc1)
+		assert.Equal(t, cc1, dbm.ReadChainConfig(hash1))
+		assert.NotEqual(t, cc2, dbm.ReadChainConfig(hash1))
+
+		dbm.WriteChainConfig(hash1, cc2)
+		assert.NotEqual(t, cc1, dbm.ReadChainConfig(hash1))
+		assert.Equal(t, cc2, dbm.ReadChainConfig(hash1))
+	}
+}
+
+// TestDBManager_Preimage tests read/write operations of preimages.
+func TestDBManager_Preimage(t *testing.T) {
+	for _, dbm := range dbManagers {
+		assert.Nil(t, nil, dbm.ReadPreimage(hash1))
+
+		preimages1 := map[common.Hash][]byte{hash1: hash2[:], hash2: hash1[:]}
+		dbm.WritePreimages(num1, preimages1)
+
+		assert.Equal(t, hash2[:], dbm.ReadPreimage(hash1))
+		assert.Equal(t, hash1[:], dbm.ReadPreimage(hash2))
+
+		preimages2 := map[common.Hash][]byte{hash1: hash1[:], hash2: hash2[:]}
+		dbm.WritePreimages(num1, preimages2)
+
+		assert.Equal(t, hash1[:], dbm.ReadPreimage(hash1))
+		assert.Equal(t, hash2[:], dbm.ReadPreimage(hash2))
+	}
+}
+
+// TestDBManager_MainChain tests service chain related database operations, used in main chain.
```

Comment:
```suggestion
// TestDBManager_ParentChain tests service chain related database operations, used in parent chain.
```
Please apply this modification. :)